spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraReader.scala

```diff
@@ -0,0 +1,166 @@
+package com.datastax.spark.connector.rdd
+
+import java.io.IOException
+
+import com.datastax.driver.core.{ProtocolVersion, Session}
+import com.datastax.spark.connector._
+import com.datastax.spark.connector.cql.{CassandraConnector, Schema}
+import com.datastax.spark.connector.rdd.reader._
+
+import scala.reflect.ClassTag
+
+trait CassandraReader[R] {
```

Comment:
This needs better documentation what it does. Seems like it deals mostly with checking schema / columns, but doesn't do any *reading*. Can you think of a better name?